libopenage/pathfinding/pathfinder.cpp

```diff
@@ -266,20 +270,18 @@ const Pathfinder::portal_star_t Pathfinder::portal_a_star(const PathRequest &req
 		}
 
 		// get the exits of the current node
-		auto exits = current_node->get_exits(visited_portals, current_node->entry_sector);
+		auto& exits = current_node->exits[current_node->entry_sector];
 
 		// evaluate all neighbors of the current candidate for further progress
-		for (auto &exit : exits) {
-			if (exit->was_best) {
+		for (auto &[exit_id, distance_cost] : exits) {
+			auto& exit = portal_map.at(exit_id);
+			exit->entry_sector = current_node->portal->get_exit_sector(current_node->entry_sector);
+			if (was_best.find(exit->portal->get_id()) != was_best.end()) {
 				continue;
 			}
 
-			// Get distance cost (from current node to exit node)
-			auto distance_cost = Pathfinder::distance_cost(
-				current_node->portal->get_exit_center(current_node->entry_sector),
-				exit->portal->get_entry_center(exit->entry_sector));
 
-			bool not_visited = !visited_portals.contains(exit->portal->get_id());
+			bool not_visited = visited_portals.find(exit->portal->get_id()) == visited_portals.end(); 
```

Comment:
The change is probably not faster if you don't use the result of `find`, so I would change it back for readability.

Reply:
i've changed visited_portals to be of type unordered_set, find seems to be the best available method.

Reply:
unordered_set supports `contains()` https://en.cppreference.com/w/cpp/container/unordered_set/contains